internal/policy/container/has_modified_files.go

```diff
@@ -180,69 +202,242 @@ func (p HasModifiedFilesCheck) Metadata() check.Metadata {
 	}
 }
 
-// Untar takes a destination path and a reader; a tar reader loops over the tarfile
-// creating the file structure at 'dst' along the way, and writing any files
-func untar(pathChan chan<- string, r io.Reader) error {
-	tr := tar.NewReader(r)
+func extractPackageNameVersionRelease(pkgList []*rpmdb.PackageInfo) map[string]struct{} {
+	pkgNameList := make(map[string]struct{}, len(pkgList))
+	for _, pkg := range pkgList {
+		pkgNameList[fmt.Sprintf("%s-%s-%s", pkg.Name, pkg.Version, pkg.Release)] = struct{}{}
+	}
+	return pkgNameList
+}
 
-	for {
-		header, err := tr.Next()
+// FindRPMDB attempts to extract a valid RPMDB from layers in the order
+// they are provided. If found is not set to true, foundIndex and pkglist should
+// be disregarded as any value there will be invalid.
+func findRPMDB(ctx context.Context, layer v1.Layer) (found bool, pkglist []*rpmdb.PackageInfo) {
+	logger := logr.FromContextOrDiscard(ctx)
+	var err error
+	pkglist, err = extractRPMDB(ctx, layer)
+	if err == nil {
+		id, _ := layer.Digest()
+		logger.V(log.TRC).Info("findRPMDB found an RPM db", "layer", id)
+		found = true
+		return
+	}
 
-		switch {
-		// if no more files are found return
-		case err == io.EOF:
-			close(pathChan)
-			return nil
+	return found, pkglist
+}
 
-		// return any other error
-		case err != nil:
-			return err
+// DirectoryIsExcluded excludes a directory and any file contained in that directory.
+func directoryIsExcluded(ctx context.Context, s string) bool {
+	excl := map[string]struct{}{
+		"etc": {},
+		"var": {},
+		"run": {},
+	}
 
-		// if the header is nil, just skip it (not sure how this happens)
-		case header == nil:
-			continue
+	for k := range excl {
+		if strings.HasPrefix(s, filepath.Clean(k+"/")) || k == s {
```

Comment:
Is the `+"/"` necessary? Won't clean take care of the case where the trailing `/` is left off? Maybe I read clean incorrectly. Or you are trying to force a slash to insure clean doesn't error, since `//` will be taken care of anyways.

Reply:
@komish wrote this part. IIUC, it's so that no matter what, at least '/' will be there. If k was cleaned completely, it would return '.'

Reply:
Confirmed ^

Reply:
Alternatively, could we use `normalize`